libsolidity/analysis/ReferencesResolver.h

```diff
@@ -43,12 +43,25 @@ class ReferencesResolver: private ASTConstVisitor
 {
 public:
 	ReferencesResolver(
-		ASTNode& _root,
+		ErrorList& _errors,
 		NameAndTypeResolver& _resolver,
 		ContractDefinition const* _currentContract,
 		ParameterList const* _returnParameters,
 		bool _resolveInsideCode = false
-	);
+	):
+		m_errors(_errors),
+		m_resolver(_resolver),
+		m_currentContract(_currentContract),
+		m_returnParameters(_returnParameters),
+		m_resolveInsideCode(_resolveInsideCode)
+	{}
+
+	/// @returns true if no errors during resolving
+	bool resolve(ASTNode& _root)
+	{
+		_root.accept(*this);
+		return Error::containsOnlyWarnings(m_errors);
```

Comment:
There could already be errors in `m_errors` from completely unrelated parts of the compiler.
Please use a flag that is set in `typeError()` and reset here.